### DISCUSSION TITLE
Grafana provisioning

### DISCUSSION THREAD

**Discussion by longzheng** [jasonacox/Powerwall-Dashboard#491] [General]
@jasonacox I'm just curious what your thoughts are on Grafana provisioning.

I saw that there was a [pull request a few months ago](https://github.com/jasonacox/Powerwall-Dashboard/issues/461) that disabled the env variable (and I believe it was never implemented).

I played around with it a bit and it was definitely possible to get the Influx data source automatically set up. 

The sun and moon data source is a bit tricker since it requires lat/lon, but I was thinking perhaps a script could automatically generate it with a free geoip API like https://freeipapi.com/api/json which automatically returns the lat/lon based on the requesting IP.

Curious what your thoughts are.

**Comment by jasonacox** [comment]
Hi @longzheng - I always thought it would be a nice enhancement and make the last step of the setup easier. Any downsides?

If you or anyone else would like to work on this, please do. Happy to test and accept a PR if someone has time to work on it.

**Comment by longzheng** [comment]
Addressed in https://github.com/jasonacox/Powerwall-Dashboard/pull/512
